src/transformers/models/mask2former/modeling_mask2former.py

```diff
@@ -926,7 +926,7 @@ def forward(
         encoder_attention_mask=None,
         position_embeddings: Optional[torch.Tensor] = None,
         reference_points=None,
-        spatial_shapes=None,
+        spatial_shapes_list=None,
```

Comment:
changing the name is a breaking change, we should probably have a deprecation cycle no?

Reply:
if not, we can add 🔴 as I think the motivation is strong enough.

Reply:
Hmm, I suppose it is an internal module of the model, not sure if it is intended to be used elsewhere, let me know if I'm wrong

Reply:
Isn't changing the logic, but keeping both `spatial_shapes` and `spatial_shapes_list` already a breaking change?

Sure there's some question of whether users can rely on internal Modules of transformers models, but also if a user doesn't pass a value for `spatial_shapes_list` this code will fail as [L939](https://github.com/huggingface/transformers/pull/34393/files#diff-1fee617cdf6a827988f42ee7a5cde812e08a4867257e03684df091e03a4141deR939) will try to iterate over a None object.

BTW seems like the changes in #33600 already violate this contract([see this line](https://github.com/huggingface/transformers/pull/33600/files#diff-137bda311c71a6ad1765c311a04366dc1b547213c95795752ddb11e39466bbabR711))? I followed that PR as a guide on what I should change here.

It seems like the proper way forward is to add a 🔴  here.

Reply:
What's the path forward @ArthurZucker ?

Reply:
Bumping this again after the weekend. I'm not familiar with Huggingface's policies of 
1. What qualifies as a breaking change?
2. What the release process is?

Could you provide a recommendation on where we should take this PR?

Reply:
Usually even if a module is `Internal` people still end up using it 😅 
This IS a breaking change, but acceptable IMO. Let's use  some  🚨 on the PR title to make sure we communicate about it on the release!